development/lib/retry.js

```diff
@@ -46,18 +46,22 @@ async function retry(
         console.error('error caught in retry():', error);
       }
 
-      if (attempts < retries) {
-        console.log('Ready to retry() again');
+      if (retryUntilFailure) {
+        throw new Error('Test failed. No more retries will be performed');
       }
 
-      if (retryUntilFailure) {
-        return null;
+      if (attempts < retries) {
+        console.log('Ready to retry() again');
       }
     } finally {
       attempts += 1;
     }
   }
 
+  if (retryUntilFailure) {
+    return null;
+  }
+
```

Comment:
before it assumed that if we reach the max retries, it was an error and throw the error with rejection message `'Retry limit reached'`, but in the case of using retryUntilFailure, reaching the max retry limit it's actually a good thing, as it means the test has not failed. That's why in this case I'm returning just null

Reply:
Can we rename this parameter `stopAfterOneFailure` for better clarity?

Reply:
changed :+1: